### PR TITLE
View, create, and edit config maps

### PR DESCRIPTION
**Pull request by spadgett**
https://trello.com/c/gUzfL5OA

Config map list:

screen shot 2016-11-14 at 2 58 42 pm

Browse config map:

screen shot 2016-11-10 at 4 38 00 pm

Create config map:

screen shot 2016-11-14 at 2 59 09 pm

CC @jwforres

### CODE REVIEW

File: app/scripts/controllers/configMap.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        DataService
+          .get("configmaps", $routeParams.configMap, context)
+          .then(function(configMap) {
+            configMapResolved(configMap);
+            watches.push(DataService.watchObject("configmaps", $routeParams.configMap, context, configMapResolved));
+          }, function(e) {
+            $scope.loaded = true;
+            $scope.alerts["load"] = {
+              type: "error",
+              message: "The route details could not be loaded.",
```
Comment (jwforres) [inline app/scripts/controllers/configMap.js]: The config map details

File: app/scripts/controllers/otherResources.js [hunk 2]
```diff
@@ -21,6 +21,7 @@ angular.module('openshiftConsole')
         case "DeploymentConfig":
         case "BuildConfig":
         case "Build":
+        case "ConfigMap":
```
Comment (jwforres) [inline app/scripts/controllers/otherResources.js]: while you are in here, can you add Secret to this list

File: app/views/directives/edit-config-map.html [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+          A unique key for this config map entry.
+        </div>
+        <div class="has-error" ng-show="configMapForm['key-' + $id].$error.oscUnique && configMapForm['key-' + $id].$touched">
+          <span class="help-block">
+            Duplicate key "{{item.key}}". Keys must be unique within the config map.
+          </span>
+        </div>
+        <div class="has-error" ng-show="configMapForm['key-' + $id].$error.pattern && configMapForm['key-' + $id].$touched">
+          <span class="help-block">
+            Config map keys may only consist of letters, numbers, periods, and underscores.
```
Comment (jwforres) [inline app/views/directives/edit-config-map.html]: hyphens?

**Comment by jwforres** [comment]
changes mostly look good, couple minor things.  I'll pull this down tomorrow and try it out.

**Comment by spadgett** [comment]
@jwforres updated

**Comment by jwforres** [comment]
On the config map details page, the labels dont show up

**Comment by spadgett** [comment]
> On the config map details page, the labels dont show up

Thanks, fixed.

**Comment by jwforres** [comment]
[merge]

**Pull request merged**